Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift

```diff
@@ -289,6 +289,12 @@ public struct ApolloCodegenConfiguration: Codable, Equatable {
     ///
     /// See `APQConfig` for more information on Automatic Persisted Queries.
     public let apqs: APQConfig
+    /// Annotate generated Swift code with the Swift `available` attribute and `deprecated`
+    /// argument for parts of the GraphQL schema annotated with the built-in `@deprecated`
+    /// directive.
+    ///
+    /// This is currently limited to enum values.
```

Comment:
This implementation is currently limited to only deprecated enum values but if it can be extended to the other parts of a GraphQL schema that can be deprecated we should consider it.

Reply:
> We should 100% de doing deprecations warnings for deprecated fields as well if this is set to true.

To be done in a separate PR.